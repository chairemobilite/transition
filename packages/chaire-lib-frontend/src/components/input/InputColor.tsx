/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import React from 'react';
import { CirclePicker } from 'react-color';

import { _isBlank } from 'chaire-lib-common/lib/utils/LodashExtensions';
import Preferences from 'chaire-lib-common/lib/config/Preferences';

export type InputColorProps = {
    id: string;
    onValueChange: (e: any) => void;
    /** Hex string representing the default color. For example: #123456 */
    defaultColor: string;
    /** Hex string representing the color. For example: #123456 */
    value?: string;
};

type InputColorState = {
    displayColorPicker: boolean;
};

class InputColor extends React.Component<InputColorProps, InputColorState> {
    constructor(props: InputColorProps) {
        super(props);

        this.state = {
            displayColorPicker: false
        };

        this.handleColorChange = this.handleColorChange.bind(this);
        this.handleOpenColorPicker = this.handleOpenColorPicker.bind(this);
        this.handleCloseColorPicker = this.handleCloseColorPicker.bind(this);
    }

    handleOpenColorPicker(e: React.MouseEvent): void {
        e.preventDefault();
        this.setState({ displayColorPicker: true });
    }

    handleCloseColorPicker(): void {
        this.setState({ displayColorPicker: false });
    }

    handleColorChange(color, _e: React.MouseEvent): void {
        this.props.onValueChange({ target: { value: color.hex } });
    }

    render(): React.ReactNode {
        const color = !_isBlank(this.props.value) ? this.props.value : this.props.defaultColor;

        const popover = {
            position: 'absolute',
            zIndex: 2
        } as React.CSSProperties;

        const cover = {
            position: 'fixed',
            top: '0px',
            right: '0px',
            bottom: '0px',
            left: '0px'
        } as React.CSSProperties;

        return (
            <div className="_input _blank">
                <div
                    className="_circle-button _open-color-picker"
                    style={{ backgroundColor: color }}
                    onClick={this.handleOpenColorPicker}
                ></div>
                {this.state.displayColorPicker && (
                    <div style={popover}>
                        <div style={cover} onClick={this.handleCloseColorPicker} className="_close-color-picker" />
                        <div
                            className="_input-color"
                            style={{ backgroundColor: '#000000', border: 'none', padding: '0.5rem' }}
                        >
                            <CirclePicker /*disableAlpha={true}*/
                                circleSize={18}
                                circleSpacing={8}
                                color={color}
                                colors={Preferences.current.colorPicker.colors}
                                onChangeComplete={this.handleColorChange}
                            />
                        </div>
                    </div>
                )}
            </div>
        );
    }
}
export default InputColor;
