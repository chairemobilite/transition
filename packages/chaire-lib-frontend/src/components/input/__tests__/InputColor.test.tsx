/*
 * Copyright 2022, Polytechnique Montreal and contributors
 *
 * This file is licensed under the MIT License.
 * License text available at https://opensource.org/licenses/MIT
 */
import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import InputColor from '../InputColor';

const mockOnChange = jest.fn();
const testId = 'ColorWidgetId';

test('Default props', () => {
    const { container } = render(<InputColor
        id = {testId}
        onValueChange = {mockOnChange}
        defaultColor = '#554433'
    />);
    expect(container.firstChild).toMatchSnapshot();
});

test('All props', () => {
    const { container } = render(<InputColor
        id = {testId}
        onValueChange = {mockOnChange}
        value = '#123456'
        defaultColor = '#123456'
    />);
    expect(container.firstChild).toMatchSnapshot();
});

test('Blank value', () => {
    const { container } = render(<InputColor
        id = {testId}
        onValueChange = {mockOnChange}
        value = ''
        defaultColor = '#223344'
    />);
    expect(container.firstChild).toMatchSnapshot();
});

test('With colorpicker', () => {
    const { container } = render(<InputColor
        id = {testId}
        onValueChange = {mockOnChange}
        value = '#123456'
        defaultColor = '#223344'
    />);
    const colorPickerOpenButton = container.querySelector('._open-color-picker') as HTMLInputElement;
    fireEvent.click(colorPickerOpenButton);
    expect(container.firstChild).toMatchSnapshot();

    const colorPickerCloseButton = container.querySelector('._close-color-picker') as HTMLInputElement;
    fireEvent.click(colorPickerCloseButton);
    expect(container.firstChild).toMatchSnapshot();
});
