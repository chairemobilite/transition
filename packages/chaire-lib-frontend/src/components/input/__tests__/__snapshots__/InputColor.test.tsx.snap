// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All props 1`] = `
<div
  class="_input _blank"
>
  <div
    class="_circle-button _open-color-picker"
    style="background-color: rgb(18, 52, 86);"
  />
</div>
`;

exports[`Blank value 1`] = `
<div
  class="_input _blank"
>
  <div
    class="_circle-button _open-color-picker"
    style="background-color: rgb(34, 51, 68);"
  />
</div>
`;

exports[`Default props 1`] = `
<div
  class="_input _blank"
>
  <div
    class="_circle-button _open-color-picker"
    style="background-color: rgb(85, 68, 51);"
  />
</div>
`;

exports[`With colorpicker 1`] = `
<div
  class="_input _blank"
>
  <div
    class="_circle-button _open-color-picker"
    style="background-color: rgb(18, 52, 86);"
  />
  <div
    style="position: absolute; z-index: 2;"
  >
    <div
      class="_close-color-picker"
      style="position: fixed; top: 0px; right: 0px; bottom: 0px; left: 0px;"
    />
    <div
      class="_input-color"
      style="background-color: rgb(0, 0, 0); padding: 0.5rem;"
    >
      <div
        class="circle-picker "
        style="width: 252px; display: flex; flex-wrap: wrap; margin-right: -8px; margin-bottom: -8px;"
      >
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #0086FF; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#0086FF"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #00ffdd; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#00ffdd"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #00ff55; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#00ff55"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #aaff00; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#aaff00"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #fff600; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#fff600"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff9400; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff9400"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff3f00; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff3f00"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff0000; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff0000"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff0061; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff0061"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff00c3; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff00c3"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #d000ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#d000ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #7b00ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#7b00ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #4800ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#4800ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #1d00ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#1d00ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #004787; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#004787"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #008270; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#008270"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #00842c; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#00842c"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #5a8700; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#5a8700"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #878200; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#878200"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #824b00; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#824b00"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #872100; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#872100"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #840000; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#840000"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #910037; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#910037"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #870067; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#870067"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #6e0087; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#6e0087"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #410087; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#410087"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #250084; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#250084"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #0f0087; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#0f0087"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #7cc2ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#7cc2ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #99fff1; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#99fff1"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #8effb4; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#8effb4"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #d3ff7c; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#d3ff7c"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #fffa7f; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#fffa7f"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ffc677; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ffc677"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff9977; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff9977"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ff89b6; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ff89b6"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #e98cff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#e98cff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #aa89ff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#aa89ff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #ffffff; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#ffffff"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #c9c9c9; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#c9c9c9"
              />
            </span>
          </div>
        </span>
        <span>
          <div
            style="width: 18px; height: 18px; margin-right: 8px; margin-bottom: 8px; transform: scale(1); transition: 100ms transform ease;"
          >
            <span>
              <div
                style="background: transparent; height: 100%; width: 100%; cursor: pointer; position: relative; outline: none; border-radius: 50%; box-shadow: inset 0 0 0 10px #878787; transition: 100ms box-shadow ease;"
                tabindex="0"
                title="#878787"
              />
            </span>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`With colorpicker 2`] = `
<div
  class="_input _blank"
>
  <div
    class="_circle-button _open-color-picker"
    style="background-color: rgb(18, 52, 86);"
  />
</div>
`;
